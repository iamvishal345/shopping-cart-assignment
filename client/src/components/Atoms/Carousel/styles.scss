.car-container {
  margin-top: 2rem;
  .banner {
    display: flex;
    justify-content: center;
    position: relative;
    background-color: white;
    .overlay {
      position: absolute;
      width: 100%;
      height: 100%;
      display: flex;
      align-items: center;
      justify-content: space-between;
      .car-btn {
        padding: 8px;
        border: none;
        color: white;
        background-color: black;
        opacity: 0.4;
      }
      .current {
        display: flex;
        align-self: flex-end;
        margin-bottom: 1rem;
        .dot {
          height: 5px;
          width: 5px;
          background-color: #79716f;
          border-radius: 50%;
          margin: 0 8px;
          &.active {
            background-color: black;
          }
        }
      }
    }
    &:after {
      content: "";
      position: absolute;
      z-index: -1;
      bottom: 10px;
      left: 10px;
      width: 98%;
      height: 15%;
      border-radius: 50%;
      box-shadow: 0 15px 10px rgb(0 0 0 / 70%);
    }
  }
}
