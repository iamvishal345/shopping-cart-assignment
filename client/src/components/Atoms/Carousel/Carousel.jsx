import React, { useState } from "react";
import "./styles.scss";
export const Carousel = ({ items }) => {
  const [curr, setCurr] = useState(0);

  // useEffect(() => {
  //   const interval = setInterval(() => {
  //     console.log("called", curr);
  //     if (curr + 1 > items.length - 1) {
  //       setCurr(0);
  //     } else {
  //       setCurr(curr + 1);
  //     }
  //   }, 5000);
  //   return () => {
  //     clearInterval(interval);
  //   };
  // }, [curr, items]);

  const handleClick = (i) => {
    if (curr + i > items.length - 1) {
      setCurr(0);
    } else if (curr + i < 0) {
      setCurr(items.length - 1);
    } else {
      setCurr(curr + i);
    }
  };
  return (
    <div className="car-container">
      <div className="banner">
        <img
          loading="lazy"
          height="auto"
          width="100%"
          src={process.env.PUBLIC_URL + items[curr].bannerImageUrl}
          alt={items[curr].bannerImageAlt}
        />
        <div className="overlay">
          <button
            id="prev-button"
            className="car-btn"
            onClick={(e) => handleClick(-1)}
          >
            PREV
          </button>
          <div className="current">
            {items.map((item, i) => (
              <div
                key={item.bannerImageUrl}
                className={`dot ${curr === i ? "active" : ""}`}
              ></div>
            ))}
          </div>
          <button
            id="next-button"
            className="car-btn"
            onClick={(e) => handleClick(1)}
          >
            NEXT
          </button>
        </div>
      </div>
    </div>
  );
};
