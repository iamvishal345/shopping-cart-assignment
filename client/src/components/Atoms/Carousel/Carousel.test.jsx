import React from "react";
import { shallow } from "enzyme";
import { Carousel } from "./Carousel";

const items = [
  {
    bannerImageUrl: "/static/images/offers/offer1.jpg",
    bannerImageAlt: "Independence Day Deal - 25% off on shampoo",
    order: 1,
    id: "5b6c38336cb7d770b7010ccd",
  },
  {
    bannerImageUrl: "/static/images/offers/offer2.jpg",
    bannerImageAlt: "Independence Day Deal - Rs120 off on surf",
    order: 2,
    id: "5b6c38156cb7d770b7010ccc",
  },
  {
    bannerImageUrl: "/static/images/offers/offer3.jpg",
    bannerImageAlt: "Independence Day Deal - Rs99 off on Rice",
    order: 3,
    id: "5b6c38156cb7d770b7010cce",
  },
];

describe("carousel", () => {
  let wrapped;
  beforeEach(() => {
    wrapped = shallow(<Carousel items={items} />);
  });
  it("should match the snapshot", () => {
    expect(wrapped).toMatchSnapshot();
  });

  it("should render the carousel with image of first item", () => {
    expect(wrapped.find("img").prop("alt")).toEqual(items[0].bannerImageAlt);
    expect(wrapped.find("img").prop("src")).toEqual(items[0].bannerImageUrl);
    expect(wrapped.find(".dot").at(0).prop("className")).toEqual("dot active");
  });
  it("should show next item on click of next button", () => {
    wrapped.find("#next-button").simulate("click", {});
    expect(wrapped.find("img").prop("alt")).toEqual(items[1].bannerImageAlt);
    expect(wrapped.find("img").prop("src")).toEqual(items[1].bannerImageUrl);
    expect(wrapped.find(".dot").at(1).prop("className")).toEqual("dot active");
  });
  it("should show prev item on click of prev button", () => {
    wrapped.find("#prev-button").simulate("click", {}); //From the first item it will go to last
    expect(wrapped.find("img").prop("alt")).toEqual(items[2].bannerImageAlt);
    expect(wrapped.find("img").prop("src")).toEqual(items[2].bannerImageUrl);
    expect(wrapped.find(".dot").at(2).prop("className")).toEqual("dot active");
    wrapped.find("#next-button").simulate("click", {}); //From the last on click of next it goes to first item
    expect(wrapped.find("img").prop("alt")).toEqual(items[0].bannerImageAlt);
    expect(wrapped.find("img").prop("src")).toEqual(items[0].bannerImageUrl);
    expect(wrapped.find(".dot").at(0).prop("className")).toEqual("dot active");
  });
});
