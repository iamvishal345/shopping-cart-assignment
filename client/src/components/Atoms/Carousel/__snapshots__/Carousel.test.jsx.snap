// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`carousel should match the snapshot 1`] = `
<div
  className="car-container"
>
  <div
    className="banner"
  >
    <img
      alt="Independence Day Deal - 25% off on shampoo"
      height="auto"
      loading="lazy"
      src="/static/images/offers/offer1.jpg"
      width="100%"
    />
    <div
      className="overlay"
    >
      <button
        className="car-btn"
        id="prev-button"
        onClick={[Function]}
      >
        PREV
      </button>
      <div
        className="current"
      >
        <div
          className="dot active"
          key="/static/images/offers/offer1.jpg"
        />
        <div
          className="dot "
          key="/static/images/offers/offer2.jpg"
        />
        <div
          className="dot "
          key="/static/images/offers/offer3.jpg"
        />
      </div>
      <button
        className="car-btn"
        id="next-button"
        onClick={[Function]}
      >
        NEXT
      </button>
    </div>
  </div>
</div>
`;
